net/dnstest/dnstest.go

```diff
@@ -0,0 +1,109 @@
+package dnstest
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/miekg/dns"
```

Comment:
I've also looked at https://github.com/foxcpp/go-mockdns which is also based on `github.com/miekg/dns` but decided not to pull-in intermediate dependency. I also think it has startup race which I try to avoid using `NotifyStartedFunc` (see `startServer`)

Reply:
That's great Miekg know what he does and interaction was always great.